### ISSUE TITLE
fail to run typechain on Andre Cronje rarity.sol

### ISSUE THREAD

**Issue by StErMi** [ethereum-ts/TypeChain#497]
Hi there, I've tried to use typechain generate types of Andre Cronje [rarity.sol](https://github.com/andrecronje/rarity/blob/main/core/rarity.sol) contract but I'm getting this error:

```
SyntaxError: Parameter declaration expected. (476:64)
  474 |       
  475 |
> 476 |     'summoned(address,uint256,uint256)'(owner?: string | null, class?: null, summoner?: null, ): TypedEventFilter<[string, BigNumber, BigNumber], {owner: string,class: BigNumber,summoner: BigNumber }>;
      |                                                                ^
  477 |     
  478 |       summoned(owner?: string | null, class?: null, summoner?: null, ): TypedEventFilter<[string, BigNumber, BigNumber], {owner: string,class: BigNumber,summoner: BigNumber }>;
  479 |       
    at e (/Users/stermi/dev/web3/rarity-achievements/node_modules/typechain/node_modules/prettier/parser-typescript.js:1:322)
    at /Users/stermi/dev/web3/rarity-achievements/node_modules/typechain/node_modules/prettier/parser-typescript.js:22:3328347
    at Object.parse (/Users/stermi/dev/web3/rarity-achievements/node_modules/typechain/node_modules/prettier/parser-typescript.js:22:3328380)
    at Object.parse$a [as parse] (/Users/stermi/dev/web3/rarity-achievements/node_modules/typechain/node_modules/prettier/index.js:13729:19)
    at coreFormat (/Users/stermi/dev/web3/rarity-achievements/node_modules/typechain/node_modules/prettier/index.js:15260:16)
    at formatWithCursor$1 (/Users/stermi/dev/web3/rarity-achievements/node_modules/typechain/node_modules/prettier/index.js:15500:14)
    at /Users/stermi/dev/web3/rarity-achievements/node_modules/typechain/node_modules/prettier/index.js:60458:12
    at Object.format (/Users/stermi/dev/web3/rarity-achievements/node_modules/typechain/node_modules/prettier/index.js:60478:12)
    at prettierOutputTransformer (/Users/stermi/dev/web3/rarity-achievements/node_modules/typechain/src/codegen/outputTransformers/prettier.ts:8:19)
    at /Users/stermi/dev/web3/rarity-achievements/node_modules/typechain/src/typechain/io.ts:22:33 {
  loc: { start: { line: 476, column: 64 } },
  codeFrame: '\x1B[0m \x1B[90m 474 |\x1B[39m       \x1B[0m\n' +
    '\x1B[0m \x1B[90m 475 |\x1B[39m\x1B[0m\n' +
    "\x1B[0m\x1B[31m\x1B[1m>\x1B[22m\x1B[39m\x1B[90m 476 |\x1B[39m     \x1B[32m'summoned(address,uint256,uint256)'\x1B[39m(owner\x1B[33m?\x1B[39m\x1B[33m:\x1B[39m string \x1B[33m|\x1B[39m \x1B[36mnull\x1B[39m\x1B[33m,\x1B[39m \x1B[36mclass\x1B[39m\x1B[33m?\x1B[39m\x1B[33m:\x1B[39m \x1B[36mnull\x1B[39m\x1B[33m,\x1B[39m summoner\x1B[33m?\x1B[39m\x1B[33m:\x1B[39m \x1B[36mnull\x1B[39m\x1B[33m,\x1B[39m )\x1B[33m:\x1B[39m \x1B[33mTypedEventFilter\x1B[39m\x1B[33m<\x1B[39m[string\x1B[33m,\x1B[39m \x1B[33mBigNumber\x1B[39m\x1B[33m,\x1B[39m \x1B[33mBigNumber\x1B[39m]\x1B[33m,\x1B[39m {owner\x1B[33m:\x1B[39m string\x1B[33m,\x1B[39m\x1B[36mclass\x1B[39m\x1B[33m:\x1B[39m \x1B[33mBigNumber\x1B[39m\x1B[33m,\x1B[39msummoner\x1B[33m:\x1B[39m \x1B[33mBigNumber\x1B[39m }\x1B[33m>\x1B[39m\x1B[33m;\x1B[39m\x1B[0m\n" +
    '\x1B[0m \x1B[90m     |\x1B[39m                                                                \x1B[31m\x1B[1m^\x1B[22m\x1B[39m\x1B[0m\n' +
    '\x1B[0m \x1B[90m 477 |\x1B[39m     \x1B[0m\n' +
    '\x1B[0m \x1B[90m 478 |\x1B[39m       summoned(owner\x1B[33m?\x1B[39m\x1B[33m:\x1B[39m string \x1B[33m|\x1B[39m \x1B[36mnull\x1B[39m\x1B[33m,\x1B[39m \x1B[36mclass\x1B[39m\x1B[33m?\x1B[39m\x1B[33m:\x1B[39m \x1B[36mnull\x1B[39m\x1B[33m,\x1B[39m summoner\x1B[33m?\x1B[39m\x1B[33m:\x1B[39m \x1B[36mnull\x1B[39m\x1B[33m,\x1B[39m )\x1B[33m:\x1B[39m \x1B[33mTypedEventFilter\x1B[39m\x1B[33m<\x1B[39m[string\x1B[33m,\x1B[39m \x1B[33mBigNumber\x1B[39m\x1B[33m,\x1B[39m \x1B[33mBigNumber\x1B[39m]\x1B[33m,\x1B[39m {owner\x1B[33m:\x1B[39m string\x1B[33m,\x1B[39m\x1B[36mclass\x1B[39m\x1B[33m:\x1B[39m \x1B[33mBigNumber\x1B[39m\x1B[33m,\x1B[39msummoner\x1B[33m:\x1B[39m \x1B[33mBigNumber\x1B[39m }\x1B[33m>\x1B[39m\x1B[33m;\x1B[39m\x1B[0m\n' +
    '\x1B[0m \x1B[90m 479 |\x1B[39m       \x1B[0m'
}
```

I think that the problem is because of this: `event summoned(address indexed owner, uint256 class, uint256 summoner);` and in typescript/js the `class` is a reserved keyword.

how could that be fixed?

**Comment by krzkaczor** [comment]
@StErMi quick fix would be to rename `class` in abi. We will roll out a proper fix soon.

**Comment by hasparus** [comment]
Hey @StErMi, this should be fixed after you upgrade to [`typechain@5.2.0`](https://github.com/dethcrypto/TypeChain/releases/tag/typechain%405.2.0).
